src/dash-table/components/Table/index.tsx

```diff
@@ -49,24 +50,45 @@ export default class Table extends Component<SanitizedAndDerivedProps, Standalon
     }
 
     componentWillReceiveProps(nextProps: SanitizedAndDerivedProps) {
-        if (nextProps.filter_query === this.props.filter_query) {
-            return;
-        }
-
         this.setState(state => {
-            const { workFilter: { map: currentMap, value } } = state;
-
-            if (value !== nextProps.filter_query) {
-                const map = this.filterMap(
-                    currentMap,
-                    nextProps.filter_query,
-                    nextProps.visibleColumns
-                );
-
-                return map !== currentMap ? { workFilter: { map, value} } : null;
-            } else {
-                return null;
+            const { applyFocus: currentApplyFocus, workFilter: { map: currentMap, value } } = state;
+
+            const nextState: Partial<StandaloneState> = {};
+
+            // state for filter
+            if (nextProps.filter_query !== this.props.filter_query) {
+                if (value !== nextProps.filter_query) {
+                    const map = this.filterMap(
+                        currentMap,
+                        nextProps.filter_query,
+                        nextProps.visibleColumns
+                    );
+
+                    if (map !== currentMap) {
+                        nextState.workFilter = { map, value };
+                    }
+                }
+            }
+
+            // state for applying focus
+            if (nextProps.active_cell !== this.props.active_cell) {
```

Comment:
what if `active_cell` was cleared? Maybe this is OK because there's nothing that `applyFocus` would actually focus *on*?

Reply:
Exactly. No `active_cell`, nothing to focus.